src/gl_hsv_thresholder.cpp

```diff
@@ -94,9 +95,10 @@ GLuint make_program(const char *vertex_source, const char *fragment_source) {
     return program;
 }
 
-GlHsvThresholder::GlHsvThresholder(int width, int height)
+GlHsvThresholder::GlHsvThresholder(int width, int height, CameraModel model)
     : m_width(width), m_height(height) {
     m_status = createHeadless();
+    useGrayScalePassThrough = isGrayScale(model);
```

Comment:
nit: initialize this as part of the top block. IE, `: m_width(width), m_height(height), useGrayScalePassThrough{isGrayScale(model)}`